sled-agent/src/services.rs

```diff
@@ -586,8 +589,8 @@ impl Task {
     }
 }
 
-/// Describes the state of a sled-local zone.
-enum SledLocalZone {
+/// Describes the state of a sled-local switch zone.
+enum SledLocalSwitchZoneState {
```

Comment:
I didn't know the reasoning behind "SledLocal", so I left it. Otherwise I kinda just wanted to name it `SwitchZoneState`

Reply:
I think the idea here was "a zone that sled-agent can locally decide whether to start" (as opposed to most control plane zones which it only starts at the request of RSS or Nexus). But since the switch zone is the only one, I'd be on board with naming this `SwitchZoneState`; if/when we have other "local" zones, they may or may not have the same state machine anyway?

Reply:
Yeah, @jgallagher is right, it was the split of "who controls this zone -- is the sled deciding to launch it locally, or is it being told to launch it?"

I like `SwitchZoneState`, I think it captures the exact info we're trying to convey. I *hope* that we don't need any more locally-launched zones like the switch zone, since the model where "Nexus controls what runs" is easier to manage, when we can use it.

Reply:
Thanks for the background information about the naming! I'll change these then